app/models/course/assessment/submission.rb

```diff
@@ -169,6 +169,20 @@ def grade
     latest_answers.map { |a| a.grade || 0 }.sum
   end
 
+  def last_graded_at
+    current_graded_at = nil
+    answers.each do |answer|
+      if answer.graded_at && (current_graded_at.nil? || current_graded_at < answer.graded_at)
+        current_graded_at = answer.graded_at
+      end
+    end
+    current_graded_at
+  end
+
+  def maximum_grade
```

Comment:
assessment has `maximum_grade`, I think you might want `submission.assessment.maximum_grade`? And if you does this, also need to eager load assessments.

Reply:
We will pass in assessment instead of using assessment from the submission.